optimum/quanto/nn/qmodule.py

```diff
@@ -200,13 +201,15 @@ def from_module(
         activations: Optional[qtype] = None,
         optimizer: Optional[Optimizer] = None,
     ):
-        qmodule = cls.qcreate(module, weights, activations, optimizer)
+        with init_empty_weights():
+            qmodule = cls.qcreate(module, weights, activations, optimizer)
         if qmodule is None:
             return None
         with torch.no_grad():
-            qmodule.weight.copy_(module.weight)
+            qmodule.weight = module.weight
```

Comment:
Not entirely sure if the copy is necessary. The copy_ call uses extra RAM if module.weight is lazily loaded (i.e. module.weight hasn't been loaded yet but will load because of the copy_ call) which has caused my computer to run out of memory in the past (i.e. loading 12B model with 32 GB RAM)